src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -175,7 +175,7 @@ public void addRooms(int num) {
         return roomList.asUnmodifiableObservableList();
     }
 
-    // TODO: remove this method and use getRoomList() instead
+    // TODO: remove this method and use getRoomList() instead (I will need this modifableRoomList for editing though)
```

Comment:
Hmm if we modify `RoomList` directly from `EditRoomCommand`, we may be breaking a few abstraction barriers since `Command` will be interacting with `RoomList`.

To avoid this problem, we can add a `setRoom(Room target, Room editedRoom)` method to `RoomList`. Then, add another `setRoom(Room target, Room editedRoom)` method to `Model` (the interface). Finally, In `ModelManager`, call `roomList.setRoom(...)`.

We can then use the `setRoom(...)` method by calling `model.setRoom(...)`. So we are restricting the interactions between classes such that `Command` only interacts with `RoomList` via `Model`.

Reply:
Wait, I see that `setRoom` has been done. Good job! In that case, why is the modifiable room list still needed?

Reply:
because i have to check if the room is present in the ObservableList

Reply:
`getRoomList()` returns the `ObservableList` though. Wouldn't it be possible to check if the room is present from there? 

I think one reason we would not want to return the `RoomList` is because it is mutable. That means that if we return `RoomList` from `ModelManager`, any other class can modify the content through `RoomList` (e.g. call `roomList.addRooms(...)`) instead of through the `Model` (i.e. call `model.addRooms(...)`).

If we